lib/tapioca/generators/base.rb

```diff
@@ -10,13 +10,18 @@ class Base
       extend T::Sig
       extend T::Helpers
 
+      class FileWriter < Thor
+        include Thor::Actions
+      end
+
       # TODO: Remove me when logging logic has been abstracted
       include Thor::Base
 
       abstract!
 
-      sig { params(default_command: String).void }
-      def initialize(default_command:)
+      sig { params(default_command: String, file_writer: Thor::Actions).void }
+      def initialize(default_command:, file_writer: FileWriter.new)
```

Comment:
You'll need to reflect this change in all subclasses initializers.

Reply:
It's using the default value which is a new instance of `FileWriter` which shouldn't require any changes to existing subclasses.